app/config/taxRatesAndBands.scala

```diff
@@ -47,15 +48,27 @@ object TaxRatesAndBands {
 
   val allRates: List[TaxRatesAndBands] = TaxRatesAndBands20152016 :: TaxRatesAndBands20162017 :: TaxRatesAndBands20172018 ::
     TaxRatesAndBands20182019 :: TaxRatesAndBands20192020 :: TaxRatesAndBands20202021 :: TaxRatesAndBands20212022 ::
-    TaxRatesAndBands20222023 :: TaxRatesAndBands20232024 :: TaxRatesAndBands20242025 :: Nil
+    TaxRatesAndBands20222023 :: TaxRatesAndBands20232024 :: TaxRatesAndBands20242025 :: TaxRatesAndBands20242025MidYearChange :: Nil
 
   val liveTaxRates: List[TaxRatesAndBands] = if (LocalDate.now.isBefore(latestTaxYearGoLiveDate)) allRates.dropRight(1) else allRates
 
-  def getRates(year: Int): TaxRatesAndBands = liveTaxRates.filter(_.taxYear == year) match {
-    case params if params.nonEmpty => params.head
+  def getRates(year: Int, disposalDate : Option[LocalDate] = None , isMidYearChangeApplicable : Boolean = false): TaxRatesAndBands = liveTaxRates.filter(_.taxYear == year) match {
+    case params if params.nonEmpty => getTaxRatesAndBandsForMidYearChange(params, disposalDate,isMidYearChangeApplicable)
     case _ => liveTaxRates.maxBy(_.taxYear)
   }
 
+  private def getTaxRatesAndBandsForMidYearChange(liveTaxRates : List[TaxRatesAndBands],disposalDate: Option[LocalDate], isMidYearChangeApplicable: Boolean): TaxRatesAndBands = {
+    if (isMidYearChangeApplicable && liveTaxRates.size > 1) {
+      liveTaxRates.filter(_.effectiveDate.nonEmpty).head match {
+        case changedTaxRates if disposalDate.nonEmpty && changedTaxRates.effectiveDate.get.isBefore(disposalDate.get.plusDays(1))  => changedTaxRates
+        case _ if disposalDate.isEmpty => throw new RuntimeException("Disposal date can not be empty")
+        case _ => liveTaxRates.filter(_.effectiveDate.isEmpty).head
```

Comment:
do we want/need to filter twice?

Reply:
Needed, as they are filters based on different condition. But, can be handled in a simpler way